Client/Frontend/Browser/TabManager.swift

```diff
@@ -679,6 +679,29 @@ class TabManager: NSObject, FeatureFlagsProtocol {
         assert(Thread.isMainThread)
         configuration.processPool = WKProcessPool()
     }
+    
+    /// When all history gets deleted, we use this special way to handle Tab History deletion. To make it appear like
+    /// the currently open tab also has its history deleted, we close the tab and reload that URL in a new tab.
```

Comment:
nit: maybe add the part where this is due to a limitation on iOS WKWebview where clearing history isn't a straight forward task